src/lib.rs

```diff
@@ -55,21 +55,21 @@ pub enum RegClass {
 /// register 0 is different than Float register 0.
 ///
 /// Because of bit-packed encodings throughout the implementation,
-/// `hw_enc` must fit in 5 bits, i.e., at most 32 registers per class.
+/// `hw_enc` must fit in 6 bits, i.e., at most 64 registers per class.
 ///
 /// The value returned by `index()`, in contrast, is in a single index
 /// space shared by all classes, in order to enable uniform reasoning
 /// about physical registers. This is done by putting the class bit at
-/// the MSB, or equivalently, declaring that indices 0..31 are the 32
-/// integer registers and indices 32..63 are the 32 float registers.
+/// the MSB, or equivalently, declaring that indices 0..63 are the 64
```

Comment:
Pre-existing but a pedantic nit on my own comment: could you adjust to either `0..=63` or `0..64` (and likewise for the second range)? Inconsistency w.r.t. Rust open/cosed-range syntax is bothering me :-)